composer/trainer/trainer.py

```diff
@@ -1691,6 +1766,40 @@ def _train_loop(self) -> None:
         self.engine.run_event(Event.FIT_END)
         self._run_evaluators(Event.FIT_END, log_level=LogLevel.FIT)
 
+    def _eval_train_metrics(self):
+        assert self._train_data_spec is not None, 'The train data spec should be set on __init__ or fit()'
+        assert self.train_metrics is not None, 'The train metrics should be set on __init__ or fit()'
+
+        self.state.model.eval()
+        with torch.no_grad():
+            # Cache the device batch, because `self.state.batch` gets overridden in microbatching loop
+            device_batch = self.state.batch
```

Comment:
the `device_batch` isn't used below, maybe cache the `device_batch_size` instead?

Reply:
oops! forgot to use it